From e46d455201d08725687dc90d3d0ee99fe8f70ca6 Mon Sep 17 00:00:00 2001
From: Frode Nordahl <frode.nordahl@canonical.com>
Date: Mon, 21 Aug 2023 15:53:33 +0200
Subject: [PATCH] docs: Add `nowarn` region option to tables.

Starting with groff 1.23.0 a warning is produced if the tbl
preprocessor is not run.  A side effect of enabling it is that
new warnings on table formatting is printed.

As requested during the review [0] of a series [1] attempting to
address this, this patch makes use of the `nowarn` region option
as opposed to attempting to change the formatting.

0: https://patchwork.ozlabs.org/project/openvswitch/patch/ZM00Wfa80rOb2oCA@riva.ucam.org/#3164177
1: https://patchwork.ozlabs.org/project/openvswitch/list/?series=367378&state=*

Reported-by: Lucas Nussbaum <lucas@debian.org>
Reported-at: https://bugs.debian.org/1042358
Signed-off-by: Frode Nordahl <frode.nordahl@canonical.com>
Signed-off-by: Ilya Maximets <i.maximets@ovn.org>
---
 build-aux/extract-ofp-fields | 6 +++---
 lib/meta-flow.xml            | 1 +
 2 files changed, 4 insertions(+), 3 deletions(-)

diff -ur openvswitch-2.17.7.orig/build-aux/extract-ofp-fields openvswitch-2.17.7/build-aux/extract-ofp-fields
--- openvswitch-2.17.7.orig/build-aux/extract-ofp-fields	2023-06-27 14:30:44.000000000 +0000
+++ openvswitch-2.17.7/build-aux/extract-ofp-fields	2025-01-22 13:58:34.757155550 +0000
@@ -578,7 +578,7 @@
     body += [""".PP
 \\fB%s Field\\fR
 .TS
-tab(;);
+tab(;),nowarn;
 l lx.
 """ % title]
 
@@ -655,7 +655,7 @@
         '.SH \"%s\"\n' % build.nroff.text_to_nroff(title.upper() + " FIELDS"),
         '.SS "Summary:"\n',
         '.TS\n',
-        'tab(;);\n',
+        'tab(;),nowarn;\n',
         'l l l l l l l.\n',
         'Name;Bytes;Mask;RW?;Prereqs;NXM/OXM Support\n',
         '\_;\_;\_;\_;\_;\_\n']
@@ -665,7 +665,7 @@
     return ''.join(content)
 
 def make_oxm_classes_xml(document):
-    s = '''tab(;);
+    s = '''tab(;),nowarn;
 l l l.
 Prefix;Vendor;Class
 \_;\_;\_
diff -ur openvswitch-2.17.7.orig/lib/meta-flow.xml openvswitch-2.17.7/lib/meta-flow.xml
--- openvswitch-2.17.7.orig/lib/meta-flow.xml	2023-06-27 14:30:44.000000000 +0000
+++ openvswitch-2.17.7/lib/meta-flow.xml	2025-01-22 13:58:45.365148939 +0000
@@ -3517,6 +3517,7 @@
     </p>
 
     <tbl>
+nowarn;
 r r r r r.
 Criteria        OpenFlow 1.0    OpenFlow 1.1    OpenFlow 1.2+   NXM
 \_      \_      \_      \_      \_
